### PR TITLE
feat(datasource): supports a new datasource of doris

### LINKED ISSUE #1448: [Feature]: Support new datasource of Doris
### Is your feature request related to a problem?

Support new datasource of Doris

### Describe the solution you'd like

Support adding Doris as a separate data source to the ODC data source, facilitating customization of Doris data source-related functionalities.

Support adding Doris schema to the project.

Support viewing Doris databases, tables, and table information in the SQL window, enabling data queries through the SQL window.

Doris adapts to work order functionalities:
a. Supports export.
b. Supports exporting result sets.
c. Supports SQL work orders (DDL, DML).
d. Supports SQL plans.

Doris adapts to security capabilities:
a. Supports 8 built-in SQL window validation rules.
b. Supports 3 built-in DML-related SQL check rules (invalid WHERE condition, no WHERE condition, syntax error).
c. Supports library-level permission validation.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ISADBA**
#### What type of PR is this?
type-feature

Optionally add one or more of the following kinds if applicable:
datasource

#### What this PR does / why we need it:
Support new datasource of Doris

#### Which issue(s) this PR fixes:
Fixes #1448

#### Special notes for your reviewer:
have frontend pr

#### Additional documentation e.g., usage docs, etc.:

```docs
Support adding Doris as a separate data source to the ODC data source, facilitating customization of Doris data source-related functionalities.

Support adding Doris schema to the project.

Support viewing Doris databases, tables, and table information in the SQL window, enabling data queries through the SQL window.

Doris adapts to work order functionalities:
a. Supports export.
b. Supports exporting result sets.
c. Supports SQL work orders (DDL, DML).
d. Supports SQL plans.

Doris adapts to security capabilities:
a. Supports 8 built-in SQL window validation rules.
b. Supports 3 built-in DML-related SQL check rules (invalid WHERE condition, no WHERE condition, syntax error).
c. Supports library-level permission validation.
```

### CODE REVIEW

**Comment by LuckyPickleZZ** [comment]
Thanks a lot for your contribution! 
I'm the developer of task-plugin-mysql, now I have roughly reviewed your code and I believe that most of the code in task-plugin-doris overlaps with MysqlDataTransferExtension and related classes.
May I ask what is the specific difference? If there is no difference, you can try to extend MySQLDataTransferExtension to implement the datatransfer function of the Doris data source.

File: server/odc-test/src/main/java/com/oceanbase/odc/test/database/TestDBConfigurations.java [hunk 1]
```diff
@@ -80,6 +103,10 @@ public TestDBConfiguration getTestMysqlConfiguration() {
         return connectType2ConfigurationMap.get(TestDBType.MYSQL);
     }
 
+    public TestDBConfiguration getTestDorisConfiguration() {
```
Comment (smallsheeeep) [inline server/odc-test/src/main/java/com/oceanbase/odc/test/database/TestDBConfigurations.java:106]: Thanks for contributing code! We will soon be supporting the Doris database in unit tests for GitHub Action.

File: server/odc-core/src/main/java/com/oceanbase/odc/core/sql/execute/mapper/DefaultJdbcRowMapper.java [hunk 2]
```diff
@@ -61,6 +61,13 @@ public DefaultJdbcRowMapper(@NonNull ConnectionSession session) {
                     ConnectionSessionUtil.getNlsTimestampFormat(session)));
             mapperList.add(new OracleNumberMapper());
             mapperList.add(new OracleBinaryNumberMapper());
+        } else if (Objects.nonNull(dialectType) && dialectType.isDoris()) {
+            mapperList.add(new MySQLBitMapper());
+            mapperList.add(new MySQLDatetimeMapper());
+            mapperList.add(new MySQLYearMapper());
+            mapperList.add(new MySQLTimestampMapper());
```
Comment (PeachThinking) [inline server/odc-core/src/main/java/com/oceanbase/odc/core/sql/execute/mapper/DefaultJdbcRowMapper.java:68]: Thanks for your contributions! After a careful review, we've noticed that this section of the code appears to share similarities with an existing conditional statement : 
`if (Objects.nonNull(dialectType) && dialectType.isMysql()).`
 To maintain the code's efficiency and clarity, we suggest that consolidate these conditions into a single unified branch like ：
`if (Objects.nonNull(dialectType) && (dialectType.isMysql()) || dialectType.isDoris())`. 
If other codes have similar situations, you can consider modifying them together～

File: server/plugins/connect-plugin-doris/src/main/java/com/oceanbase/odc/plugin/connect/doris/DorisJdbcUrlParser.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+/**
+ * ClassName: DorisJdbcUrlParser Package: com.oceanbase.odc.plugin.connect.doris Description:
+ *
+ * @Author: fenghao
+ * @Create 2024/1/4 17:35
+ * @Version 1.0
+ */
+@Slf4j
+public class DorisJdbcUrlParser implements JdbcUrlParser {
```
Comment (yhilmare) [inline server/plugins/connect-plugin-doris/src/main/java/com/oceanbase/odc/plugin/connect/doris/DorisJdbcUrlParser.java:42]: duplicated codes, OBMySQLJdbcUrlParser can replace this

File: server/plugins/connect-plugin-doris/src/main/java/com/oceanbase/odc/plugin/connect/doris/DorisDiagnoseExtensionPoint.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+ *
+ * @Author: fenghao
+ * @Create 2024/1/4 17:12
+ * @Version 1.0
+ */
+@Slf4j
+@Extension
+public class DorisDiagnoseExtensionPoint implements SqlDiagnoseExtensionPoint {
+    @Override
+    public SqlExplain getExplain(Statement statement, @NonNull String sql) throws SQLException {
```
Comment (yhilmare) [inline server/plugins/connect-plugin-doris/src/main/java/com/oceanbase/odc/plugin/connect/doris/DorisDiagnoseExtensionPoint.java:53]: It seems like that this is duplicated code

File: server/plugins/connect-plugin-doris/src/main/java/com/oceanbase/odc/plugin/connect/doris/initializer/EnableProfileInitializer.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author jingtian
+ * @date 2023/5/30
+ * @since ODC_release_4.2.0
+ */
+@Slf4j
+public class EnableProfileInitializer implements ConnectionInitializer {
```
Comment (yhilmare) [inline server/plugins/connect-plugin-doris/src/main/java/com/oceanbase/odc/plugin/connect/doris/initializer/EnableProfileInitializer.java:32]: duplicated code

File: server/plugins/task-plugin-doris/src/main/java/com/oceanbase/odc/plugin/task/doris/DorisTaskPlugin.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
+/*
```
Comment (yhilmare) [inline server/plugins/task-plugin-doris/src/main/java/com/oceanbase/odc/plugin/task/doris/DorisTaskPlugin.java:1]: dup codes

File: server/odc-core/src/main/java/com/oceanbase/odc/core/sql/execute/mapper/DefaultJdbcRowMapper.java (hunk 2 above)
Comment (ISADBA) [reply server/odc-core/src/main/java/com/oceanbase/odc/core/sql/execute/mapper/DefaultJdbcRowMapper.java:68]: Doris have owner column types, like boolean, HyperLogLog, STRUCT......

**Pull request closed without merging**